bundles/http/Citrix/AdobeSign/31fff7a5-ffa8-4019-9e2f-e477837f92de.js

```diff
@@ -0,0 +1,596 @@
+const moment = library.load('moment-timezone');
+const uuid = library.load('uuid')
+
+function getStatus(status) {
+    return status.split('_').reduce((initialValue, currnetValue) => {
+        return initialValue + (currnetValue[0].toUpperCase() + (currnetValue.toLowerCase()).substring(1, currnetValue
+            .length) + ' ')
+    }, '').trimEnd()
+}
+
+async function syncUsers(dataStore, client, integrationParameters) {
+    let userids = []
+    const PAGE_SIZE = 100
+    let cursor = ""
+    const activeUsers = []
+    let url = `/api/rest/v6/groups/${integrationParameters.GroupId}/users?pageSize=${PAGE_SIZE}`
+    do {
+        const response = await client.fetch(url);
+        const json = await response.json();
+
+        if (!response.ok) {
+            throw new Error(JSON.stringify(json))
+        }
+        const data = json.userInfoList ?? []
+        cursor = json.page.nextCursor ?? null
+        url = `/api/rest/v6/groups/${integrationParameters.GroupId}/users?pageSize=${PAGE_SIZE}&cursor=${cursor}`
+        for (const user of data) {
+            const userDetails = client.fetchSync(`api/rest/v6/users/${user.id}`).jsonSync()
+            if (userDetails.status === "ACTIVE") {
+                activeUsers.push(user)
+            }
+        }
+        let users = activeUsers.map(user => {
+            userids.push(user.id);
+            return {
+                "ID": user.id,
+                "email": user.email,
+                "company": user.company,
+                "firstName": user.firstName,
+                "lastName": user.lastName,
+                "accountId": user.accountId
+            }
+        })
+        dataStore.save("Users", users);
```

Comment:
1. Please no `fetchSync` & `jsonSync`. We don't want to stop the execution thread until it is finished despite the fact that we don't do any operation in parallel -> use `await`
2. imo `const activeUsers = []` should be on line 27 because it won't be empty in the next cycle in `do-while`
3. Why not put "the right object" directly into `activeUsers`?
 
```suggestion
        for (const user of data) {
            const response = await client.fetch(`api/rest/v6/users/${user.id}`)
            const userDetails = await response.json()
            if (userDetails.status === "ACTIVE") {
               userids.push(user.id);
               
                activeUsers.push({
                   ID: user.id,
                   email: user.email,
                   company: user.company,
                   firstName: user.firstName,
                   lastName: user.lastName,
                   accountId: user.accountId
                })
            }
        }
        dataStore.save("Users", activeUsers);
```

Reply:
changes implemented